_docs/software_support_lifecycle.md

```diff
@@ -0,0 +1,86 @@
+
+As we operate entirely on the graces of our community members being willing to voluntarily contribute to our modules, we also need to be respectful of their time and resources and not add undue stress in order to support software that upstream creators no longer support.
+
+To account for this, we no longer support any software or operating system after the date that the maintainers have marked said software or operating system as End of Life, hereafter referred to as "EOL". However, this does not mean we will immediately remove code that allows for the module to run on EOL'd software or operating systems, but the community will no longer be expected to provide support to users for that software.
+
+There are exceptions and the full policy is detailed in the next section.
+
+### Support Lifecycle
+
+#### Operating Systems
+An Operating System is no longer considered supported once it reaches an EOL date determined by the upstream maintainer* of the operating system. To ensure our users have enough time to adjust, we will take the following steps to ensure as seamless of a transition as possible:
+
+1. A deprecation PR will be created no later that 3 months before an Operating System's EOL date.
+2. Removal of the EOL version from the metadata.json and tests to be completed by the EOL date.
+3. Issues relating to the EOL operating system will no longer be addressed after the EOL date.
+4. Code allowing a module to be run on the EOL operating system will be left in the module until one of the following milestones are reached after which time the code will be removed:
+    1. Maintaining the code prevents the module from moving forward.
+    2. 12 Months after the EOL date.
+5. Pull Requests will continue to be accepted with the following exceptions:
+    1. Accepting such a pull request would be detrimental to the module's long term goals.
+    2. The code has already been removed as per the standards set in section 4.
+
+*maintainers include, but are not limited to, the community, company, or organization that maintains ownership over a specific piece of software or project.
+
+##### Support Table:
+
+| OS Name | Support Begins | EOL | Notification PR | End of Support | Removal of Code |
+| ------- | -------------- | ----------- | --------------- | -------------- | --------------- |
+| CentOS/RHEL 6 | 2011/07/10 | 2020/11/30 | N/A | 2020/11/30 | 2021/11/30 |
+| CentOS/RHEL 7 | 2014/07/07 | 2024/06/30 | 2024/03/31 | 2024/06/30 | 2025/06/30 |
+| CentOS 8 | 2019/09/24 | 2021/12/31 | 2021/09/30 | 2021/12/31 | 2022/12/31 |
+| RHEL 8 |  2019/05/01 | 2029/05/31 | 2029/02/28 | 2029/05/31 | 2030/05/31 |
+| CentOS Stream 8 | 2022/01/01 | 2029/05/31 | 2029/02/28 | 2029/05/31 | 2030/05/31 |
```

Comment:
CentOS Stream 8 upstream EOL is in May 31, 2024 according to https://centos.org/distro-faq/#q2-what-about-the-other-releases-of-centos-linux that states it's supported until "full RHEL support phase" which is defined in https://access.redhat.com/support/policy/updates/errata/#Life_Cycle_Dates and specifically doesn't include the 5 year Maintenance Phase RHEL itself gets. There will be CentOS Stream 9 with 2 year overlap released before that according to Question 6 of the FAQ, though.